src/Microsoft.Health.Fhir.CosmosDb.Initialization/Features/Storage/ResourceManagerCollectionSetup.cs

```diff
@@ -0,0 +1,253 @@
+                containerResourceInfo);
+
+            CosmosDBSqlContainerCollection containerCollection = Database.GetCosmosDBSqlContainers();
+            ArmOperation<CosmosDBSqlContainerResource> newContainerResponse = await containerCollection.CreateOrUpdateAsync(WaitUntil.Completed, CollectionId, content, cancellationToken);
+            container = newContainerResponse.Value;
+
+            var throughput =
+                _cosmosCollectionConfiguration.Get(Core.Constants.CollectionConfigurationName)
+                    .InitialCollectionThroughput;
+
+            if (throughput.HasValue)
+            {
+                _logger.LogInformation("Updating container throughput to '{Throughput}' RUs.", throughput);
+                await container
+                    .GetCosmosDBSqlContainerThroughputSetting()
+                    .CreateOrUpdateAsync(
+                        WaitUntil.Started, // Throughput provisioning can be async
+                        new ThroughputSettingsUpdateData(
+                            Location,
+                            new ThroughputSettingsResourceInfo
+                            {
+                                Throughput = throughput,
+                            }),
+                        cancellationToken);
+            }
+        }
+        else
+        {
+            _logger.LogInformation("Collection '{CollectionId}' found.", CollectionId);
+        }
+    }
+
+    public async Task InstallStoredProcs(CancellationToken cancellationToken)
+    {
+        CosmosDBSqlContainerResource containerResponse = await Database.GetCosmosDBSqlContainers().GetAsync(CollectionId, cancellationToken);
+        CosmosDBSqlStoredProcedureCollection cosmosDbSqlStoredProcedures = containerResponse.GetCosmosDBSqlStoredProcedures();
+
+        var existing = cosmosDbSqlStoredProcedures
+            .Select(x => x.Data.Resource.StoredProcedureName)
+            .ToList();
+
+        var storedProcsNeedingInstall = _storeProceduresMetadata
+            .Where(storedProc => !existing.Contains(storedProc.FullName))
+            .ToList();
+
+        foreach (IStoredProcedureMetadata storedProc in storedProcsNeedingInstall)
+        {
+            _logger.LogInformation("Installing StoredProc '{StoredProcFullName}'.", storedProc.FullName);
+
+            var cosmosDbSqlStoredProcedureResource = new CosmosDBSqlStoredProcedureResourceInfo(storedProc.FullName)
+            {
+                Body = storedProc.ToStoredProcedureProperties().Body,
+            };
+
+            var storedProcedureCreateOrUpdateContent = new CosmosDBSqlStoredProcedureCreateOrUpdateContent(Location, cosmosDbSqlStoredProcedureResource);
+            await cosmosDbSqlStoredProcedures.CreateOrUpdateAsync(WaitUntil.Completed, storedProc.FullName, storedProcedureCreateOrUpdateContent, cancellationToken);
+        }
+    }
+
+    public async Task UpdateFhirCollectionSettingsAsync(CollectionVersion version, CancellationToken cancellationToken)
```

Comment:
Are we using version in this method?

Reply:
The version was there also before the Cosmos split (in multiple places), but was used to skip applying the Collection Indexes and Ttl if they hadn't changed.

https://github.com/microsoft/fhir-server/blob/dbd7da7f7490936d974bc0db560b41ed26d4f230/src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Versioning/CollectionUpgradeManager.cs#L46-L49

https://github.com/microsoft/fhir-server/blob/dbd7da7f7490936d974bc0db560b41ed26d4f230/src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Versioning/FhirCollectionSettingsUpdater.cs#L29-L51

Some of this moved around a little. In `main` it seems to incorrectly be applied to the SearchParameterStatus settings instead of the Collection settings which is what I moved. Open to thoughts, but it is now in only one place and for the original intent.